lua/terrortown/menus/gamemode/administration/roles.lua

```diff
@@ -64,6 +64,47 @@ function CLGAMEMODESUBMENU:Populate(parent)
         master = masterEnb,
     })
 
+    form:MakeHelp({
+        label = "help_roles_derandomize",
+    })
+
+    local masterDerand = form:MakeComboBox({
+        serverConvar = "ttt_role_derandomize_mode",
+        label = "label_roles_derandomize_mode",
+        choices = {
+            {
+                title = "label_roles_derandomize_mode_none",
+                value = 0, -- ROLE_DERAND_NONE
+            },
+            {
+                title = "label_roles_derandomize_mode_base_only",
+                value = 1, -- ROLE_DERAND_BASE_FLAG
+            },
+            {
+                title = "label_roles_derandomize_mode_sub_only",
+                value = 2, -- ROLE_DERAND_SUB_FLAG
+            },
+            {
+                title = "label_roles_derandomize_mode_base_and_sub",
+                value = bit.bor(1, 2), -- ROLE_DERAND_BASE_FLAG | ROLE_DERAND_SUB_FLAG
+            },
+        },
```

Comment:
Gameplay related:
Do you have any real use cases to support those rather niche configurations?

We generally try to minimize our configuration values nowadays because they are mostly overwhelming for users and i cannot think of some cases where i would even want to use any other than the `Base + Sub` or `None` configuration.

If you don't, I propose we just reduce this to a simple On/Off toggle :)

Reply:
I personally think that `Base only` is the most valuable option. The problem described in the PR description is primarily in the I/T/other distinction, which are base roles. To be honest, I added support for subroles as something of an afterthought, realizing that it would be trivial to do and may be beneficial depending on the server's set of installed roles.

Reply:
In general I agree with @saibotk here. However this is handled as a single setting in a dropdown menu. So from my point of view this implementation is alright.